app/components/post_textbox/post_textbox_base.js

```diff
@@ -396,17 +396,68 @@ export default class PostTextBoxBase extends PureComponent {
         }
     };
 
-    handlePostDraftSelectionChanged = (event) => {
-        const cursorPosition = event.nativeEvent.selection.end;
+    handlePostDraftSelectionChanged = (event, fromHandleTextChange) => {
+        let cursorPosition;
+
+        // When called from PasteableTextInput's onSelectionChange, nativeEvent gets passsed with
+        // updated cursorPosition. When called fromHandleTextChange, there's no nativeEvent passed but
+        // cursorPosition is already updated and available in state.
+        if (fromHandleTextChange) {
+            cursorPosition = this.state.cursorPosition;
+        } else {
+            cursorPosition = event.nativeEvent.selection.end;
+        }
+
         const {cursorPositionEvent} = this.props;
 
         if (cursorPositionEvent) {
             EventEmitter.emit(cursorPositionEvent, cursorPosition);
         }
 
-        this.setState({
-            cursorPosition,
-        });
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            // Matches all text between a pair of triple backticks ```, inclusive
```

Comment:
This could go into a separate, testable function, called from `handlePostDraftSelectionChanged` when relevant. Perhaps more descriptively named `switchKeyboardForCodeBlocks`? This could reduce the need to rely on explanatory comments.

Reply:
I separated to ` switchKeyboardForCodeBlocks` and removed extra comments. Thanks for the rec!